third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -1264,14 +1264,15 @@ impl<'env> Generator<'env> {
                     } else {
                         ReferenceKind::Mutable
                     };
-                    return self.gen_borrow_field_for_unpack_ref(id, str, arg, temps, ref_kind);
+                    self.gen_borrow_field_for_unpack_ref(id, str, arg, temps, ref_kind);
```

Comment:
Was the bug here really that the loop on 1276 wasn't run for this case?

Reply:
There are two issues, one is the loop wasn't run. The other one is that the type of struct pattern should be & instead of the struct type when unpacking a reference of a struct, which is fixed in the exp_builder.